TLM/TLM/Custom/PathFinding/CustomPathManager.cs

```diff
@@ -273,96 +274,109 @@ public class CustomPathManager : PathManager {
-        /// <param name="laneTypes">allowed lane types</param>
-        /// <param name="vehicleTypes">allowed vehicle types</param>
-        /// <param name="otherLaneTypes">allowed lane types for secondary lane</param>
-        /// <param name="otherVehicleTypes">other vehicle types for secondary lane</param>
-        /// <param name="allowTransport">public transport allowed?</param>
-        /// <param name="allowUnderground">underground position allowed?</param>
-        /// <param name="position">resulting path position</param>
-        /// <returns><code>true</code> if a position could be found, <code>false</code> otherwise</returns>
-        public static bool FindCitizenPathPosition(Vector3 pos,
-                                                   NetInfo.LaneType laneTypes,
-                                                   VehicleInfo.VehicleType vehicleTypes,
-                                                   NetInfo.LaneType otherLaneTypes,
-                                                   VehicleInfo.VehicleType otherVehicleTypes,
-                                                   bool allowTransport,
-                                                   bool allowUnderground,
-                                                   out PathUnit.Position position) {
-            // TODO move to ExtPathManager after harmony upgrade
-            position = default(PathUnit.Position);
-            float minDist = 1E+10f;
-            if (ExtPathManager.Instance.FindPathPositionWithSpiralLoop(
-                    pos,
-                    ItemClass.Service.Road,
-                    laneTypes,
-                    vehicleTypes,
-                    otherLaneTypes,
-                    otherVehicleTypes,
-                    allowUnderground,
-                    false,
-                    Options.parkingAI
-                        ? GlobalConfig.Instance.ParkingAI.MaxBuildingToPedestrianLaneDistance
-                        : 32f,
-                    out PathUnit.Position posA,
-                    out _,
-                    out float distA,
-                    out _) && distA < minDist) {
-                minDist = distA;
-                position = posA;
+        /// <param name="path"></param>
+        /// <param name="startPosA"></param>
+        /// <param name="startPosB"></param>
+        /// <param name="endPosA"></param>
+        /// <param name="endPosB"></param>
+        /// <param name="vehicleType"></param>
+        /// <param name="skipQueue"></param>
+        /// <returns>bool</returns>
+        public bool CreateTransportLinePath(
+                out uint path,
+                PathUnit.Position startPosA,
+                PathUnit.Position startPosB,
+                PathUnit.Position endPosA,
+                PathUnit.Position endPosB,
+                VehicleInfo.VehicleType vehicleType,
+                bool skipQueue) {
+
+            PathCreationArgs args = default;
```

Comment:
I hope the compiler is smart enough to merge `args = default` and the following field initializations. Rider would also suggest using `args = PathCreationArgs { field = ..., field = ... }` syntax